### PR TITLE
Added tags functionality to ;plot solved

### PR DESCRIPTION
**Pull request by robert-30**

### CODE REVIEW

**Comment by robert-30** [comment]
implements #74

File: tle/cogs/graphs.py [hunk 1]
```diff
@@ -72,18 +72,22 @@ def _plot_rating(resp, mark='o', labels: List[str] = None):
     _plot_rating_bg()
 
 
-def _filter_solved_submissions(submissions, contests):
+def _filter_solved_submissions(submissions, contests, *args):
```
Comment (algmyr) [inline tle/cogs/graphs.py]: Shouldn't the last argument just be something like `tags = []`, not `*args`?

File: tle/cogs/graphs.py [hunk 2]
```diff
@@ -72,18 +72,22 @@ def _plot_rating(resp, mark='o', labels: List[str] = None):
...
+    to filter problems. If a problem is solved multiple times the first accepted
     submission is kept. The unique id for a problem is (problem name, contest start time).
     """
     submissions.sort(key=lambda sub: sub.creationTimeSeconds)
     contest_id_map = {contest.id: contest for contest in contests}
     problems = set()
     solved_subs = []
+    tags = []
+    if (len(args) > 0):
+        tags = args[0]
```
Comment (algmyr) [inline tle/cogs/graphs.py]: This can be entirely removed if the signature is fixed as described above.

File: tle/cogs/graphs.py [hunk 3]
```diff
@@ -72,18 +72,22 @@ def _plot_rating(resp, mark='o', labels: List[str] = None):
...
     for submission in submissions:
         contest = contest_id_map.get(submission.problem.contestId)
-        if submission.verdict == 'OK' and submission.problem.rating and contest:
+        if submission.verdict == 'OK' and submission.problem.rating and contest and (tags == [] or submission.problem.tag_matches(tags)):
```
Comment (algmyr) [inline tle/cogs/graphs.py]: Do `not tags`. It will work also in the case where a tuple is given as tags.

**Comment by robert-30** [comment]
Updated

**Pull request merged**